### PR TITLE
feat(ChatMessage): add read status indicator slot

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes: https://github.com/microsoft/fluent-ui-react/issues/1303
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds `readStatusIndicator` slot for `ChatMessage`:

Screenshot 2020-08-18 at 15 29 08

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts
Comment (pompomon) [inline packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts:15]: Please why do we need it like that?

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts:15]: Resolved

File: packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx:183]: Why isn't this just a `title` in the `readStatusIndicator` shorthand? I don't think we ever expose props from some shorthands as first class properties of the component.

File: packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx:93]:
```suggestion
```

File: packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx
Comment (mnajdova) [inline packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx:30]: Should we have this by default as a `backgroundImage` as we do with the other components? If content or children is specified than we won't be adding it

File: packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts:11]: Did you check if this is the best approach from accessibility point of view? Usually we avoid `absolute` positioned elements if it is not really the only way to go

File: packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx
Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx:30]: Not really sure to be honest since not the default message wouldn't have it I think, just the last messages in the chat.

File: packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts:11]: Tried with screen reader on mac and it works as expected appending the `title` to the end of the message, couldn't think in another way to achieve the same ( I am basically replicating the behavior of `author` here )

File: packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx
Comment (mnajdova) [reply packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx:30]: I don't understand why that is a problem. This would just be default style. If users specify object or title inside the shorthand we would render it otherwise no. Isn't that how it should work?

File: packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts
Comment (mnajdova) [reply packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorStyles.ts:11]: Perfect, disregard my comment then :)

File: packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx]: Removed

File: packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx]: Removed

**Comment by layershifter** [comment]
Just wondering, should it be just `readIndicator` or `statusIndicator`?

**Comment by assuncaocharles** [comment]
> Just wondering, should it be just `readIndicator` or `statusIndicator`?

I am ok renaming it to `readIndicator`

File: packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx
Comment (layershifter) [reply packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx]: @mnajdova to be honest I don't think that `backgroundImage` is a good approach to go as it causes issues like this: https://github.com/microsoft/fluentui/pull/13486

**Comment by assuncaocharles** [comment]
So @layershifter @mnajdova @pompomon 

Can we go ahead and merge it?

File: packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx
Comment (pompomon) [reply packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx]: I'll also support the `icon` and not `backgroundImage` approach :) 
For me this item is more like an additional slot, so I'd rather not have anything there by default, and leave it to the consumer

@mnajdova , please WDYT?

Comment (mnajdova) [reply packages/fluentui/docs/src/examples/components/Chat/Content/ChatExampleReadStatus.shorthand.tsx]: I agree, now that the `<Icon />` component is not that heavy, I guess default icons is a better approach 👍

File: packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorVariables.ts
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorVariables.ts:9]: I would remove the `readStatusIndicator` prefix, as all these variables are for the same component.

File: packages/fluentui/react-northstar/src/components/Chat/ChatMessageReadStatusIndicator.tsx
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Chat/ChatMessageReadStatusIndicator.tsx:16]:
```suggestion
export const ChatMessageReadIndicator = compose<
```
?

File: packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorVariables.ts
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageReadStatusIndicatorVariables.ts:9]: Done

File: packages/fluentui/react-northstar/src/components/Chat/ChatMessageReadStatusIndicator.tsx
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Chat/ChatMessageReadStatusIndicator.tsx:16]: Ok

File: packages/fluentui/react-northstar/src/components/Chat/Chat.tsx
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Chat/Chat.tsx:56]:
```suggestion
    MessageReadStatus: typeof ChatMessageReadStatus;
```

File: packages/fluentui/react-northstar/src/components/Chat/Chat.tsx
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Chat/Chat.tsx:121]:
```suggestion
Chat.MessageReadStatus = ChatMessageReadStatus;
```

File: packages/fluentui/CHANGELOG.md
Comment (mnajdova) [inline packages/fluentui/CHANGELOG.md:38]: One last inconsistency, we should either remove `readIndicator` to `readStatus`, or rename `MessageReadStatus` to `MessageReadIndicator`. I'd be ok with any combination as long as we are consistent

**Review by mnajdova — APPROVED** [review]
See last comments before merging please

**Review by pompomon — APPROVED** [review]

**Pull request merged**
